Feat/lesq 1321/add unit save buttons [LESQ-1321]

njblore opened the pull request:
## Description

Music year: {owa_music_year}

- add save buttons to unit list items behind a feature flag
- add a save button to lesson listing page and update copy and layout of other buttons
- add stub save functionality to enable toggling the save buttons (doesn't actually save anything and will revert on page refresh)


## How to test

To see the save buttons you will need to add your posthog distinct id to the feature flag `teacher-save-units`, see how to do that [here](https://www.notion.so/oaknationalacademy/How-to-add-yourself-to-a-feature-flag-17c26cc4e1b18008a54adf78a962e6ba)

1. Go to https://deploy-preview-3374--oak-web-application.netlify.thenational.academy/teachers/programmes/english-secondary-ks4-aqa/units/modern-text-first-study-4896/lessons
2. With the feature flag disabled you should not see the save buttons
3. With the feature flag enabled you should see save buttons on each unit item and each optionality item which can be toggled
4. Clicking through to a lesson listing page you should see a save button in the header which can be toggled
5. This PR also intoduces the incomplete units lesson count for optionality units , see KS4 geography for an example

## Screenshots
<img width="800" alt="Screenshot 2025-04-23 at 11 17 51" src="https://github.com/user-attachments/assets/390e53d6-fa36-4758-a16d-76702a044f7a" />
<img width="300" alt="Screenshot 2025-04-23 at 11 18 16" src="https://github.com/user-attachments/assets/4a82216c-b4ee-4743-ae12-7a9c8c6813cc" />
<img width="800" alt="Screenshot 2025-04-23 at 11 35 02" src="https://github.com/user-attachments/assets/a13768f9-b8f4-4a7a-a4e4-3713e3834e82" />
<img width="800" alt="Screenshot 2025-04-23 at 11 39 55" src="https://github.com/user-attachments/assets/9cfadb6d-9cfb-4b09-90a3-ea3fd542ceda" />


## ACs
- [x]  Favouriting feature flag
- [x]  On a unit listing page, there is a bookmark button on each unit item on the unit listing page
- [x]  Update unit listing component to include the save button both desktop and mobile ([[Figma link](https://www.figma.com/design/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?m=auto&node-id=9557-24155&t=l5KFuTYdZjkP39gK-1)](https://www.figma.com/design/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?m=auto&node-id=9557-24155&t=l5KFuTYdZjkP39gK-1))
- [x]  Update optionality unit card to include save link ([[Figma link](https://www.figma.com/design/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?m=auto&node-id=9618-4306&t=l5KFuTYdZjkP39gK-1)](https://www.figma.com/design/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?m=auto&node-id=9618-4306&t=l5KFuTYdZjkP39gK-1))
- [x]  On a lesson listing page, there is a “Save” button at the top of the page
- [x]  [[Update text on share button](https://www.figma.com/design/LSjbzdvHo17XqXGBRNJjQP/Teacher-browse?node-id=8171-63984&t=NHKa7e3xssrwkmEK-1)](https://www.figma.com/design/LSjbzdvHo17XqXGBRNJjQP/Teacher-browse?node-id=8171-63984&t=NHKa7e3xssrwkmEK-1)
- [x]  Update copy on existing lesson listing buttons in header to reduce button length
- [x]  Update unit listing card layout on mobile to accomodate new button
- [x]  Move the ‘new’ tag to the save button on the lesson listing page instead of unit downloads (ensure button heights are consistent as we previously added extra padding for the new tag)

left23 reviewed (changes requested):
This looks good!

A few points:
1. The Save button on the lesson listing page could use the same treatment as the save button on the cards - where the icon alt text is empty and the aria-label is used with the unit title on the button element.
![Selection_573](https://github.com/user-attachments/assets/b1abce01-ec20-464f-a7b5-76b6c1de1d0a)

2. I wonder if removing the `cursor: pointer` from the card div would be better for user experience? Since the whole card is no longer a link, the cursor implies it should be, and it's harder now to understand where to click to go to the unit. If the cursor is a pointer only when it hovers over the unit title and the save button, maybe this make the interaction more clear. I don't think there is any harm in keeping the hover state for the entire card, if the pointer is changed. Maybe @Monika-Koziol can advise on this aspect.
![Selection_574](https://github.com/user-attachments/assets/6fccf936-4a12-44ba-b707-b958e5389284)

3. There is an awkward viewport dimension around 780px where the button's height doesn't match the other buttons
![Selection_575](https://github.com/user-attachments/assets/34cb7421-6307-40a0-ab2e-1f8cc0026e1f)

4. Optionality units incomplete lesson counts look good!

left23 commented:
Can the "link copied" text expand beyond the width of the button?
![Selection_577](https://github.com/user-attachments/assets/63e55d4e-06e0-4336-8ea3-5415d172307b)

Monika-Koziol commented:
@njblore @left23 about the accessibility point. Not sure how hard that is but could we make the unit title underline on hover? But only if a user hovers over the text and not any other parts of the card, the save button has its own states? If you use spotify try to hover over a track on a playlist: it's one line but has multiple links within it (artist, album etc). 
![Screenshot 2025-04-24 at 15 09 31](https://github.com/user-attachments/assets/035e68f2-95a2-4331-b996-9bc978893027)

njblore commented:
@left23 I have fixed the button size at tablet viewport, removed the new tag from the continue onboarding button and moved the 'Link copied' text to below the buttons. For that I had to refactor the share button to decouple the link copied text so there are a few share buttons that should be checked: on the lesson listing page, on the lesson page (there are 2 variants here behind an experiment) and lesson download success pages.

The other fixes for the cursor, underlines and alt text on the button icons will require changes in oak-components so I'll make a PR there and update when it's ready.

left23 reviewed (approved):
Hi @njblore Thanks for the work on this.
I re-tested and I can see all the changes you made are good to go.
One last thing I noticed... before I added my distinct_id to the feature flag, I see the Download Unit button is now larger than the share button
![Selection_581](https://github.com/user-attachments/assets/9b266cb4-3eb9-47e8-be18-be6e7b39a80d)

I can confirm the share button's "link copied" text still displays well in all the scenarios!

I am not around until Tuesday so I will approve this so as not to hold anything up, but also happy to look at this again next week if needed.

oak-machine-user commented:
:tada: This PR is included in version 1.736.0 :tada:

The release is available on [GitHub release](https://github.com/oaknational/Oak-Web-Application/releases/tag/v1.736.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
